tests/PHPStan/Analyser/nsrt/preg_match_shapes.php

```diff
@@ -768,3 +768,13 @@ function bug11604b (string $string): void {
 		assertType("array{0: string, 1?: ''|'XX', 2?: ''|'YY', 3?: 'ZZ'}", $matches);
 	}
 }
+
+function testLtrimDelimiter (string $string): void {
+	if (preg_match(' /(x)/', $string, $matches)) {
```

Comment:
Instead use one of the regex in above tests with spaces prepended should work?

Reply:
yes - but I prefer to keep such testcase separately

I would even prefer to use `' /x/'` but whole regex is currently not parsed into constant type.

Reply:
Yeah, so the testcase does not. Prevent regressions for the fix

Reply:
test added in https://github.com/phpstan/phpstan-src/pull/3495/commits/dd8e9634d8dad3b21988d575b6b4eab947b49f52